cicd/publish.groovy

```diff
@@ -96,22 +95,64 @@ exists=$(github-release info -u ${GITHUB_ORGANIZATION} -r ${GITHUB_REPO}  -j)
 if [ $? -gt 0 ]; then
   echo "Creating a new release in github"
   github-release -v release --user ${GITHUB_ORGANIZATION} --repo ${GITHUB_REPO} --name "z/OS Release"
+  if [ $? -eq 0 ]; then
+    echo "Release created successfully!"
+  else
+    echo "Failed to recreate release."
+    exit 1                                     
+  fi
 fi
 
 exists=$(github-release info -u ${GITHUB_ORGANIZATION} -r ${GITHUB_REPO} --tag "${TAG}" -j)
 if [ $? -gt 0 ]; then
   echo "Creating a new tag in github"
   github-release -v release --user ${GITHUB_ORGANIZATION} --repo ${GITHUB_REPO} --tag "${TAG}" --name "${NAME}" --description "${DESCRIPTION}"
+  if [ $? -eq 0 ]; then
+    echo "Release created successfully!"
+  else
+    echo "Failed to create release."
+    exit 1                                     
+  fi
 else
   echo "Deleting and creating new tag"
   github-release -v delete --user ${GITHUB_ORGANIZATION} --repo ${GITHUB_REPO} --tag "${TAG}"
   github-release -v release --user ${GITHUB_ORGANIZATION} --repo ${GITHUB_REPO} --tag "${TAG}" --name "${NAME}" --description "${DESCRIPTION}"
+  if [ $? -eq 0 ]; then
+    echo "Release recreated successfully!"
+  else
+    echo "Failed to recreate release."
+    exit 1                                     
+  fi
 fi
 
-sleep 30 # Let github register the release
-
-echo "Release should now exist"
-github-release info -u ${GITHUB_ORGANIZATION} -r ${GITHUB_REPO} --tag "${TAG}" -j
+# check up to 5 times for the release to appear on github                                                     
+for i in {1..5}; do
```

Comment:
not sure how expensive the call is but would it better to go with maybe 2 seconds and do it 25 times? 
if 10s is the 'sweet spot', then i am fine with this.

Reply:
From prior experience, it usually takes about 5-10s for the release to be reflected in github, so I'll stick with 10s for now and see how it goes.